internal/blockchain/ethereum/ethereum.go

```diff
@@ -947,3 +931,52 @@ func (e *Ethereum) GetAndConvertDeprecatedContractConfig(ctx context.Context) (l
 	})
 	return location, fromBlock, err
 }
+
+func (e *Ethereum) GetTransactionStatus(ctx context.Context, operation *core.Operation) (interface{}, error) {
+	txnID := (&core.PreparedOperation{ID: operation.ID, Namespace: operation.Namespace}).NamespacedIDString()
+
+	transactionRequestPath := fmt.Sprintf("/transactions/%s", txnID)
+	client := e.client
+	var resErr ethError
+	var statusResponse fftypes.JSONObject
+	res, err := client.R().
+		SetContext(ctx).
+		SetError(&resErr).
+		SetResult(&statusResponse).
+		Get(transactionRequestPath)
+	if err != nil || !res.IsSuccess() {
+		if res.StatusCode() == 404 {
+			return nil, nil
+		}
+		return nil, wrapError(ctx, &resErr, res, err)
+	}
+
+	headers := statusResponse.GetObject("headers")
+	receiptInfo := statusResponse.GetObject("receipt")
+	txStatus := statusResponse.GetString("status")
+
+	if txStatus != "" {
+		// If the status has changed, mock up blockchain receipt as if we'd received it
+		// as a web socket notification
+		if operation.Status == core.OpStatusPending && txStatus != ethTxStatusPending {
+			receipt := &common.BlockchainReceiptNotification{
+				Headers: common.BlockchainReceiptHeaders{
+					ReceiptID: headers.GetString("id"),
+					ReplyType: headers.GetString("type")},
+				TxHash:  statusResponse.GetString("transactionId"),
+				Message: statusResponse.GetString("errorMessage"),
+				ProtocolID: common.ProtocolIDForReceipt(
```

Comment:
I'm not sure we want a split responsibility on the generation of this.
It seems on the normal notification code path we just take this from the payload supplied by FFTM (which I think is the only connector implementation that correctly populates it today), whereas here we construct it